### ISSUE TITLE
User is not warned when Create New Repo or Rename repo functions fail

### ISSUE THREAD

**Issue by cckozie** [unfoldingWord/translationCore#6430]
2.0.0 (4b3ebee) 
ISSUE A: If the Internet connection is lost during a repo create or rename event, no alert is displayed even though the repo is not renamed or created
1. Stop and restart tC (or log out and back in) to reset the Internet will be used flag
2. Click the FAB on the Projects page and choose Import Online Project
3. Click the Access Internet button on the alert (DO NOT CHECK THE BOX)
4. Select one of the projects in your repo to download and click the Import button
5. Click the Access Internet button on the alert (DO NOT CHECK THE BOX)
6. On the Project details screen, choose a different translation id
7. On the Project Named alert, click the Create New Repository button
8. EDIT: When the accessing Internet alert is displayed, turn ~on~ off your Internet connection
9. Click the Access Internet button
10. Note that no warning was displayed about the Internet not being available
11. Navigate to your repo on Door43 and note that the requested repo was not created

ISSUE B: If the user attempts to create or rename a repo that has the same name as an existing repo, no alert is displayed even though the process has failed.
1. Import a project from your repo and change the translation id on the project details screen
2. On the project renamed alert, choose Create New Repository
3. Stop tC
4. Delete the previously imported project from your Projects folder
5. Start tC
6. Repeat step 1. above and use the same translation id as you specified before
7. EDIT: Repeat step 2. above
8. Note that no warning is displayed when the process could not be completed

**Comment by PhotoNomad0** [comment]
Notes:
- on Issue A it looks like the retry dialog is being shown on screen, but program is not waiting for the response before continuing as if everything is fine.

**Comment by PhotoNomad0** [comment]
Verified in tCore 2.1.0 (3091b6e)

**Comment by cckozie** [comment]
2.1.0 (3091b6e)
Testing of Issue B has revealed two problems. To replicate follow these steps as recording in the video [6430A.mp4.zip](https://github.com/unfoldingWord/translationCore/files/3779505/6430A.mp4.zip)


1. Start with two projects with the same language and book and different translation id's in a Door43 repo as in
<img width="221" alt="6430A-1" src="https://user-images.githubusercontent.com/8171759/67692941-e7ab6900-f96e-11e9-8408-a2fabef2691b.png">

2. Import one those projects changing the translation id to that of the other. (I imported en_tit_ultb and changed the tid to ulta)
3. Choose the Rename Local Project on the Rename Failed alert
4. On the Project Details screen change the tid to a new one unused in the repo (I used ultc)
5. Click the Save Changes button on the Project Details screen
6. DO NOT CLICK any button on the Your Project Has Been Renamed alert
7. Note that after about 7 seconds the alert is automatically dismissed. PROBLEM 1
8. When the Tools page is displayed, look at the repo on Door43 and note that both of the original projects have been deleted and the only one remaining is the one with the tid specified in step 4. PROBLEM 2
<img width="233" alt="6430A-2" src="https://user-images.githubusercontent.com/8171759/67693830-5d640480-f970-11e9-989d-ac01127fb49a.png">

**Comment by PhotoNomad0** [comment]
@cckozie 
OK - think I found the problem.  After the user changes the translation ID in the project details, tCore was going back to retry the rename.  Unfortunately it was leaving the DCS rename prompt on the screen and not showing that it was doing actually doing a DCS rename.  That Is easy enough to fix.

**Comment by PhotoNomad0** [comment]
@cckozie - after the user changes translation ID in the project details, should we re-prompt if they want to rename repo or create new repo?  Currently we just go with previous choice.

**Comment by cckozie** [comment]
@PhotoNomad0 Good question! I think we can just go with their previous choice.

**Comment by PhotoNomad0** [comment]
~@cckozie - have a test build for you to try: https://tc-builds.door43.org/build/release-v2.1.0-test.2/translationCore-macos-x64-2.1.0-dd991d3.dmg~

**Comment by PhotoNomad0** [comment]
@cckozie - another try here - have a new test build https://tc-builds.door43.org/build/release-v2.1.0-test.3/translationCore-macos-x64-2.1.0-af36189.dmg

this should also fix the problem of deleting any projects with the temporary translation IDs

**Comment by cckozie** [comment]
@PhotoNomad0 "this should also fix the problem of deleting any projects with the temporary translation IDs" I don't know what that means:)

**Comment by PhotoNomad0** [comment]
@cckozie - when you change to a translation id that is already used on DCS - that is the temporary one if  you then chose to change to another translation id (rename local).

**Comment by cckozie** [comment]
@PhotoNomad0 Following the steps in my comment yesterday, in steps 6 and 7 I'm still seeing the Your project has been renamed alert briefly appear. It feels weird to see an alert that has option buttons being automatically dismissed. How hard would it be to hide that alert when it is going to automatically be dismissed?
The improper deleting of Door43 repos has been fixed.

[6430B.mp4.zip](https://github.com/unfoldingWord/translationCore/files/3783947/6430B.mp4.zip)

**Comment by PhotoNomad0** [comment]
@cckozie - thanks will try again.   I agree it shouldn't be that hard, but it always seems to be difficult to synchronize async events.  Do you have the log file?

**Comment by cckozie** [comment]
@PhotoNomad0 
[Tue Oct 29 2019.log.zip](https://github.com/unfoldingWord/translationCore/files/3784242/Tue.Oct.29.2019.log.zip)

**Comment by cckozie** [comment]
@PhotoNomad0 I'm having second thoughts about my suggestion that we should use the user's original response to rename the Door43 repo. In the process of testing now I've found myself surprised when my repo "disappeared" cuz I had kind of forgotten I first selected the Rename Repo option, and then on the last alert I responded to I chose the Rename Local Project option. I think the word "Local" there caused me to believe it would be local only. I'd like to chat about this with you, Jesse, and Elsy after standup today.

**Comment by PhotoNomad0** [comment]
@cckozie - OK found the place where the spurious repo renamed dialog was being generated.  Also changed to return  to DCS create/rename prompt after changing translation ID.  Have new build to test:  https://tc-builds.door43.org/build/release-v2.1.0-test.4/translationCore-macos-x64-2.1.0-27a1ce3.dmg

This should also update manifest on DCS with new translation ID after rename/create new.

**Comment by cckozie** [comment]
@PhotoNomad0 Yes, I like that behavior much better!

**Comment by PhotoNomad0** [comment]
@cckozie - working on fix for case that Rich saw in testing:
```
Rich Mahn 1:45 PM
for https://github.com/unfoldingWord/translationCore/pull/6514, maybe doesn't matter, but during the 2nd to last testing step, rename should succeed this time and project opened., maybe due to my slow internet access, once I clicked the final "Rename Repository" it just showed the projects list, so I clicked on "Import Online Project" to see my online repos, THEN it popped up with the modal saying it was preparing project or whatever and went to the tools page. So for some reason had a chance to interact with the projects list in-between steps there. Doesn't break anything, and if I go back to the projects page it then shows the search dialog still there. Just wondering if it should not allow user interaction between renaming and processing/opening project though.
```

**Comment by PhotoNomad0** [comment]
@cckozie - have a test build to try:

From the `release-v2.1.0-test.4` branch:
[Linux build 2.1.0 (22ab4ef)](https://tc-builds.door43.org/build/release-v2.1.0-test.4/translationCore-linux-x64-2.1.0-22ab4ef.deb) is ready for download.
[macOS build 2.1.0 (22ab4ef)](https://tc-builds.door43.org/build/release-v2.1.0-test.4/translationCore-macos-x64-2.1.0-22ab4ef.dmg) is ready for download.
[Windows x64 build 2.1.0 (22ab4ef)](https://tc-builds.door43.org/build/release-v2.1.0-test.4/translationCore-win-x64-2.1.0-22ab4ef.setup.exe) is ready for download.
[Windows x32 build 2.1.0 (22ab4ef)](https://tc-builds.door43.org/build/release-v2.1.0-test.4/translationCore-win-x32-2.1.0-22ab4ef.setup.exe) is ready for download.

**Comment by PhotoNomad0** [comment]
Verified that Changes are in translationCore 2.1.0 (d8f36fe)

**Comment by cckozie** [comment]
Waiting on fix for #6508 in order to test this one.
Also need to test #6523 to see if it is also fixed by this one.
